### ISSUE TITLE
 word-wrap vulnerable to Regular Expression Denial of Service

### ISSUE THREAD

**Issue by vianch** [jonschlinkert/word-wrap#40]
All versions of the package word-wrap are vulnerable to Regular Expression Denial of Service (ReDoS) due to the usage of an insecure regular expression within the result variable.

![Screenshot 2023-06-28 at 12 16 27](https://github.com/jonschlinkert/word-wrap/assets/1800887/7a9172d2-e8a5-48e2-95cb-cfaced31061c)

**Comment by magnussp** [comment]
Just encountered this as well as latest [firebase-admin](https://github.com/firebase/firebase-admin-node) package is reliant on some packages that in turn is reliant on `word-wrap`.

**Comment by eclousersans** [comment]
Dealing with this as well. `eslint` depends on `optionator` which depends on `word-wrap`.

Any ETA on patch release?

**Comment by eclousersans** [comment]
Per this PR:
https://github.com/jonschlinkert/word-wrap/pull/33

They aren't able to merge in due to bus factor and the original repo owner getting hit by a bus. 

As a hack, they've published a new version under another package name. 

Par for the course in JS community...

**Comment by wellwelwel** [comment]
> Dealing with this as well. `eslint` depends on `optionator` which depends on `word-wrap`.
> 
> Any ETA on patch release?

@eclousersans, which version of ESLint do you use?
Since ESLint version **7**, it was solved in https://github.com/gkz/optionator/issues/44.

**Comment by silverbackdan** [comment]
As seen here https://github.com/jonschlinkert/word-wrap/pull/33#issuecomment-1611621997
```
"overrides": {
  "word-wrap" : "npm:@aashutoshrathi/word-wrap"
}
```
It appears this is currently a maintained version which you can override with.

**Comment by jacquesg** [comment]
This unfortunately breaks `eslint` for me:

```
Oops! Something went wrong! :(                                                                                                                                                                            
                                                                                                                                                                                                          
ESLint: 8.43.0                                                                                                                                                                                            
                                                                                                                                                                                                          
Error: Cannot find module 'word-wrap'                                                                                                                                                                     
Require stack:                                                                                                                                                                                            
- /home/jacquesg/dev/projects/lossbook/antaeus/node_modules/optionator/lib/help.js                                                                                                                        
- /home/jacquesg/dev/projects/lossbook/antaeus/node_modules/optionator/lib/index.js                                                                                                                       
- /home/jacquesg/dev/projects/lossbook/antaeus/node_modules/eslint/lib/options.js                                                                                                                         
- /home/jacquesg/dev/projects/lossbook/antaeus/node_modules/eslint/lib/cli.js                                                                                                                             
- /home/jacquesg/dev/projects/lossbook/antaeus/node_modules/eslint/bin/eslint.js                                                                                                                          
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)                                                                                                                         
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)                                                                                                                                    
    at Module.require (node:internal/modules/cjs/loader:1005:19)                                                                                                                                          
    at require (node:internal/modules/cjs/helpers:102:18)                                                                                                                                                 
    at Object.<anonymous> (/home/jacquesg/dev/projects/lossbook/antaeus/node_modules/optionator/lib/help.js:6:14)                                                                                         
    at Object.<anonymous> (/home/jacquesg/dev/projects/lossbook/antaeus/node_modules/optionator/lib/help.js:260:4)                                                                                        
    at Module._compile (node:internal/modules/cjs/loader:1105:14)                                                                                                                                         
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)                                                                                                                           
    at Module.load (node:internal/modules/cjs/loader:981:32)                                                                                                                                              
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)   
```

**Comment by silverbackdan** [comment]
I'm also on eslint 8.43.0 - is it possible there is some cache left-overs?

**Comment by jacquesg** [comment]
No, retried from clean as well. Ended up adding an override for optionator instead, which accomplishes the same thing effectively:

```
"overrides": {
  "optionator" : "0.9.3"
}
```

**Comment by silverbackdan** [comment]
Interesting - pleased you found your solution too.

**Comment by bjornjorgensen** [comment]
Any updates?

**Comment by mahnoorkazidh** [comment]
npm audit doesn't fix it for me.. unfortunately. Any update on the solution?

**Comment by larouxn** [comment]
> npm audit doesn't fix it for me.. unfortunately. Any update on the solution?

The best solution we have right now (from [here](https://github.com/jonschlinkert/word-wrap/pull/33)) is to switch to a patched fork. https://github.com/jonschlinkert/word-wrap/pull/33#issuecomment-1509703360

```javascript
"resolutions": {
  "word-wrap": "npm:@aashutoshrathi/word-wrap@^1.2.4"
},
```

Looks like `eslint` bumped their affected dependency `optionator` so that may work for you as well. https://github.com/eslint/eslint/issues/17117

```javascript
"resolutions": {
  "optionator": "^0.9.3"
}
```

Non-Yarn users can use `overrides` in place of `resolutions` apparently. (haven't tried)

Lastly, it seems many have swapped over to the fork according to the [fork's page on npmjs.org](https://www.npmjs.com/package/@aashutoshrathi/word-wrap).

![Screenshot from 2023-07-04 18-58-59](https://github.com/jonschlinkert/word-wrap/assets/1557529/ecf2620b-366a-4478-a19f-1c5299563ec5)

**Comment by xmariopereira** [comment]
The original repo owner got hit by a bus... 

Meanwhile tech people: "where is the fix!!!"

**Comment by JTBrinkmann** [comment]
> The original repo owner got hit by a bus...

@xmariopereira they were not literally hit by a bus, it's technical jargon ([bus factor](https://en.wikipedia.org/wiki/Bus_factor)). You can see some activity in PR #33, which addresses this issue.

**Comment by kachkaev** [comment]
Duplicate of https://github.com/jonschlinkert/word-wrap/issues/32

**Comment by Anmol-Baranwal** [comment]
Hi, everyone.

I was looking for an answer, and never thought it would be as simple as this.

```
npm audit fix
```
It will ultimately update the version of the dependency package and it creates no problem.

For reference of the change, you can see [here](https://github.com/Anmol-Baranwal/BabySteps-Project/commit/323e977916d9b1df39853656fb0792aaecde07d8)

**Comment by doowb** [comment]
Fixed in `word-wrap@1.2.4`.
